### PR TITLE
Cleaned up code

### PR DESCRIPTION
**Pull request by lino-levan**
- Removed a lot of unused imports
- Cleaned up some code that was technically deprecated

### CODE REVIEW

File: src/main/java/minicraft/entity/furniture/Furniture.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public Furniture(String name, Sprite sprite, int xr, int yr) {
 	@Override
 	public Furniture clone() {
 		try {
-			return getClass().newInstance();
+			return getClass().getDeclaredConstructor().newInstance();
```
Comment (Makkkkus) [inline src/main/java/minicraft/entity/furniture/Furniture.java:55]: Scary code

File: src/main/java/minicraft/core/io/Localization.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public static String getLocalized(String key) {
 		if (key.matches("^[ ]*$")) return key; // Blank, or just whitespace
 		
 		try {
-			double num = Double.parseDouble(key);
+			Double.parseDouble(key);
```
Comment (Makkkkus) [inline src/main/java/minicraft/core/io/Localization.java:51]: Is this necessary?

File: src/main/java/minicraft/entity/mob/RemotePlayer.java [hunk 3]
```diff
@@ -18,7 +17,7 @@
 import minicraft.level.Level;
 
 /** This is used for players in multiplayer mode. */
-public class RemotePlayer extends Player implements ClientTickable {
+public class RemotePlayer extends Player {
```
Comment (Makkkkus) [inline src/main/java/minicraft/entity/mob/RemotePlayer.java:20]: Are you sure this won't break something?

File: src/main/java/minicraft/saveload/LegacyLoad.java [hunk 4]
```diff
@@ -104,7 +104,6 @@ public void loadFromFile(String filename) {
 			br = new BufferedReader(new FileReader(filename));
 			
 			String curLine;StringBuilder total = new StringBuilder();
-			ArrayList<String> curData;
```
Comment (Makkkkus) [inline src/main/java/minicraft/saveload/LegacyLoad.java:107]: Please don't touch LegacyLoad. It is called that for a reason.

File: src/main/java/minicraft/screen/TitleDisplay.java [hunk 5]
```diff
@@ -23,7 +21,6 @@
 import minicraft.level.Level;
 import minicraft.screen.entry.BlankEntry;
 import minicraft.screen.entry.LinkEntry;
-import minicraft.screen.entry.ListEntry;
 import minicraft.screen.entry.SelectEntry;
```
Comment (Makkkkus) [inline src/main/java/minicraft/screen/TitleDisplay.java:24]: Good

File: src/main/java/minicraft/core/io/Localization.java (hunk 2 above)
Comment (lino-levan) [reply src/main/java/minicraft/core/io/Localization.java:51]: Technically not _necessary_, but removing unnecessary variable declarations/assignments is probably a good idea.

File: src/main/java/minicraft/entity/mob/RemotePlayer.java (hunk 3 above)
Comment (lino-levan) [reply src/main/java/minicraft/entity/mob/RemotePlayer.java:20]: Player already implements ClientTickable, so this is unneeded verbosity imo

File: src/main/java/minicraft/saveload/LegacyLoad.java (hunk 4 above)
Comment (lino-levan) [reply src/main/java/minicraft/saveload/LegacyLoad.java:107]: Fair, I will remove this

File: src/main/java/minicraft/entity/furniture/Furniture.java (hunk 1 above)
Comment (lino-levan) [reply src/main/java/minicraft/entity/furniture/Furniture.java:55]: How so? `getClass().newInstance()` skips compile-time checks of constructors. I would rather get a compile error (with `getClass().getDeclaredConstructor().newInstance()`) than a runtime error from this.

File: src/main/java/minicraft/screen/TitleDisplay.java (hunk 5 above)
Comment (lino-levan) [reply src/main/java/minicraft/screen/TitleDisplay.java:24]: :)

File: src/main/java/minicraft/entity/furniture/Furniture.java (hunk 1 above)
Comment (Makkkkus) [reply src/main/java/minicraft/entity/furniture/Furniture.java:55]: Thank you for explaining

**Pull request merged**